### PR TITLE
dialects: Add a builder for fully static memref.subview

### PR DESCRIPTION
**Pull request by PapyChacal**

### CODE REVIEW

File: xdsl/dialects/memref.py [hunk 1]
```diff
@@ -379,6 +377,49 @@ class Subview(Operation):
...
+        strides: Sequence[int],
+    ) -> Subview:
+
+        source = SSAValue.get(source)
+
+        layout_strides = [1]
+        for input_size in reversed(source_shape[1:]):
+            layout_strides.insert(0, layout_strides[0] * input_size)
+
+        layout_offset = sum([
```
Comment (superlopuh) [inline xdsl/dialects/memref.py:396]:
```suggestion
        layout_offset = sum(
```
No need to construct intermediate list

File: xdsl/dialects/memref.py [hunk 2]
```diff
@@ -379,6 +377,49 @@ class Subview(Operation):
...
+            stride * offset
+            for (stride, offset) in zip(layout_strides, offsets)
```
Comment (superlopuh) [inline xdsl/dialects/memref.py:398]:
```suggestion
            for stride, offset in zip(layout_strides, offsets)
```

Comment (georgebisbas) [reply xdsl/dialects/memref.py:398]: I feel pairs read better as it is (w parentheses)

Comment (superlopuh) [reply xdsl/dialects/memref.py:398]: Happy with either

Comment (AntonLydike) [reply xdsl/dialects/memref.py:398]: I think we go without parenthesis everywhere else in the project, so I'd go without them

**Review by AntonLydike — APPROVED** [review]
Oh my god, there it is finally! Lovely to see, looks good to me!

**Review by webmiche — APPROVED** [review]
I am not quite sure what the timeline for making the xDSL syntax obsolete is, but moving tests from `.xdsl` to `.mlir` feels dangerous as my might lose coverage on the xDSL printer/parser.

What do you think @math-fehr? Will the xdsl syntax even be part of the next release(s)?

From the Memref side, I don't see any issues :+1:

**Comment by math-fehr** [comment]
> I am not quite sure what the timeline for making the xDSL syntax obsolete is, but moving tests from `.xdsl` to `.mlir` feels dangerous as my might lose coverage on the xDSL printer/parser.
> 
> What do you think @math-fehr? Will the xdsl syntax even be part of the next release(s)?
> 
> From the Memref side, I don't see any issues +1

I feel we can already start moving to the MLIR syntax!
I'm trying to have the MLIR syntax ready before OOPSLA, so we should be feature-complete by then (probably even earlier).
So yeah, maybe the next release will already deprecate xDSL syntax?

**Pull request merged**